terragrunt/aws/cloudfront/waf.tf

```diff
@@ -0,0 +1,181 @@
+resource "aws_wafv2_web_acl" "cds_website_waf" {
+  provider = aws.us-east-1
+
+  name        = "cds_website_waf"
+  description = "WAF for CDS Website"
+  scope       = "CLOUDFRONT"
+
+  tags = {
+    CostCentre = var.billing_code
+    Terraform  = true
+  }
+
+  default_action {
+    allow {}
+  }
+
+  rule {
+    name     = "AWSManagedRulesAmazonIpReputationList"
+    priority = 10
+
+    override_action {
+      none {}
+    }
+
+
+    statement {
+      managed_rule_group_statement {
+        name        = "AWSManagedRulesAmazonIpReputationList"
+        vendor_name = "AWS"
+      }
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "AWSManagedRulesAmazonIpReputationList"
+      sampled_requests_enabled   = true
+    }
+  }
+
+  rule {
+    name     = "APIRateLimit"
```

Comment:
nit: could you change this to just something like `RateLimit` since there isn't an API.